server/bin/pbench-tarball-stats

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+"""Pbench Tar Ball Stats
+
+Scan through the ARCHIVE hierarchy for tar balls and report some
+statistics on them:
+
+  * Total number of controllers
+  * Tar balls per controller (most to least)
+  * Tar balls by month for all years, count
+  * Tar balls by month for all years, total size
+  * Top 10 most recent controllers
+  * Top 10 largest tar balls
+"""
+
+import collections
+import locale
+import os
+import re
+import sys
+
+from datetime import datetime
+from argparse import ArgumentParser
+
+import pbench
+from pbench import (
+    PbenchConfig,
+    BadConfig,
+    get_pbench_logger,
+)
+
+
+_NAME_ = "pbench-tarball-stats"
+
+tb_pat_r = (
+    r"\S+_(\d\d\d\d)[._-](\d\d)[._-](\d\d)[T_](\d\d)[._:](\d\d)[._:](\d\d)\.tar\.xz"
+)
+tb_pat = re.compile(tb_pat_r)
```

Comment:
Any real need to retain both the string and the `Pattern` object?

Reply:
No.